internal/pkg/pkg_test.go

```diff
@@ -424,6 +424,16 @@ func TestSubpackages(t *testing.T) {
 						"local-sub1",
 					},
 				},
+				{
+					matcher:   []SubpackageMatcher{None},
```

Comment:
It is good that we add tests for this. Is there a real use-case for this, or is it just for completeness? Asking for a list of subpackages and then also specify that we don't want any subpackages returned seems a bit strange.

Reply:
This would be for completeness. I agree that internally we should be smart about how we deal with copying subpackages, but we should probably make sure this utility function handles it as expected, which is to say not copying them at all.